### PR TITLE
[DFMD-156] Build a To-Do App with REACT

### PR DESCRIPTION
**Pull request by MoyinoluwaA**
Screenshot (167)
Screenshot (166)

### CODE REVIEW

**Review by ToluAkin — CHANGES_REQUESTED** [review]
@MoyinoluwaA What you have here is good but what is expected of you is have all the integration and functionality be managed in state and not use `useState` and `useEffect`

**Comment by MoyinoluwaA** [comment]
@ToluAkin I made the corrections, please review the changes

File: submissions/moyinoluwaAdenuga/todo-app/src/App.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+		this.setState({
+			todos: updatedTodos,
+			active: this.state.active - 1
+		})
+    }   
+
+   
+    // To move completed task back to todo item
+    incompleteTodo = (id) => {
+        let updatedTodos = this.state.todos.map((todo) => {
```
Comment (ToluAkin) [inline submissions/moyinoluwaAdenuga/todo-app/src/App.js:41]: Create a function for this and call it in the incomplete todo and complete todo @MoyinoluwaA

**Review by ToluAkin — CHANGES_REQUESTED** [review]
@MoyinoluwaA You are not meant to use bootstrap for the task. Kindly remove all the associated classes and import of bootstrap

File: submissions/moyinoluwaAdenuga/todo-app/src/Components/form.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        e.preventDefault()
+        
+        if (this.state.input === '') {
+            return false
+        }
+
+        this.props.onSubmit({
+            text: this.state.input, 
+            completed: false, 
+            id: Math.random() * 1000,
```
Comment (ToluAkin) [inline submissions/moyinoluwaAdenuga/todo-app/src/Components/form.js:22]: This is not good practice. You can not be setting the this way.

**Review by ToluAkin — CHANGES_REQUESTED** [review]

**Comment by MoyinoluwaA** [comment]
@ToluAkin I made the requested changes. Please review. Thanks

**Review by ToluAkin — CHANGES_REQUESTED** [review]
@MoyinoluwaA Your app is not mobile responsive, from your css, you are setting width and height. It is not in all cases you have to use both, just one and you let the content fill it.
Also, from cloning your project, the list is not displaying just the count is changing. Are you working on fixing that?

File: submissions/moyinoluwaAdenuga/todo-app/src/Components/form.js (hunk 2 above)
Comment (ToluAkin) [reply submissions/moyinoluwaAdenuga/todo-app/src/Components/form.js:22]: @MoyinoluwaA Kindly work on this

**Comment by MoyinoluwaA** [comment]
@ToluAkin I am surprised about this. It is working fine here..both mobile responsiveness and the list display. I will check back and get back to you

**Review by ToluAkin — CHANGES_REQUESTED** [review]

**Comment by ToluAkin** [comment]
This is what I have on my end @MoyinoluwaA https://monosnap.com/direct/yKvbDr7phJ6xEGD2VkmpNiZbG9bsXY

**Comment by MoyinoluwaA** [comment]
@ToluAkin or do you mean I have to add the tasks??? What I did was I made it that the users can add the task and enter it. Do you mean i should hardcode the tasks on the design??

**Comment by ToluAkin** [comment]
> @ToluAkin or do you mean I have to add the tasks??? What I did was I made it that the users can add the task and enter it. Do you mean i should hardcode the tasks on the design??

You are not meant to hard code tasks. I'm saying as a user, the tasks are not displayed after adding.

File: submissions/moyinoluwaAdenuga/todo-app/src/Components/form.js (hunk 2 above)
Comment (MoyinoluwaA) [reply submissions/moyinoluwaAdenuga/todo-app/src/Components/form.js:22]: @ToluAkin I already did. That is the previous commit. The recent one has the corrected changes

**Comment by MoyinoluwaA** [comment]
@ToluAkin I will check that out once I am able to because it was working very fine before I pushed my work. Thanks

**Comment by MoyinoluwaA** [comment]
@ToluAkin The todo app is working perfectly fine here and displaying the items. I suggest you could check it out using another means as I dont think the error is from here. Also, the onSubmit event that seemed nested and you corrected it wasn't good practise, I changed it since yesterday..I don't know why it isn't reflecting on your end. Please check the files changed for the current files i have by the last commit.
I honestly hope to be able to get this PR merged soonest. Thanks

File: submissions/moyinoluwaAdenuga/todo-app/src/Components/form.js (hunk 2 above)
Comment (ToluAkin) [reply submissions/moyinoluwaAdenuga/todo-app/src/Components/form.js:22]: Which recent ones? This branch would be updated your changes if it has been corrected.

**Review by ToluAkin — CHANGES_REQUESTED** [review]

**Comment by MoyinoluwaA** [comment]
Screenshot_20210324-101425_Chrome
Screenshot_20210324-101447_Chrome
This is what is currently there. I removed the object from the event handler...created it separately as an object and then passed it as argument into the event handler @ToluAkin

File: submissions/moyinoluwaAdenuga/todo-app/src/Components/form.js (hunk 2 above)
Comment (ToluAkin) [reply submissions/moyinoluwaAdenuga/todo-app/src/Components/form.js:22]: Okay. i see the miscommunication. Don't use `Math.Random` as Id. Use the position of the todo in an array.

**Comment by MoyinoluwaA** [comment]
Should I create a new PR? @ToluAkin Because it seems it's referencing a changed file at your end. Or any other suggestions?

**Review by ToluAkin — COMMENTED** [review]
> Should I create a new PR? @ToluAkin Because it seems it's referencing a changed file at your end. Or any other suggestions?

I only need this bit of your PR changed `id: Math.random() * 1000` Use the position of the todo and set it as the `id` as oppose to `Math.random`. The screenshot and the contents of your PR match. I don't see a reason to open a new PR.

**Comment by MoyinoluwaA** [comment]
Ok. Will do that now. Thanks.

**Comment by MoyinoluwaA** [comment]
@ToluAkin I just corrected that. Please review. Thanks

**Review by ToluAkin — CHANGES_REQUESTED** [review]
@MoyinoluwaA Well done. I noticed if I click on any of `All Items` ,`Active` or `Completed` text at the footer, the list hides and unhides. That is very usual. Kindly work on that. Also, the page is not responsive in tablet view.

**Comment by MoyinoluwaA** [comment]
@ToluAkin please I don't understand your last review. How exactly is it supposed to be when those items are clicked on??? Is there a particular way it's supposed to be in tablet mode because it is working fine here. Should I attach various screenshots of tablet screen here?

**Comment by ToluAkin** [comment]
> @ToluAkin please I don't understand your last review. How exactly is it supposed to be when those items are clicked on??? Is there a particular way it's supposed to be in tablet mode because it is working fine here. Should I attach various screenshots of tablet screen here?

Nothing is suppose to happen when they are clicked which is why I said it is unusual. `All Items` , `Active` and `Completed`, they move out of place on tablet screen unlike their position on large screen and nothing like small screen.

**Comment by MoyinoluwaA** [comment]
@ToluAkin ooh..I thought they were supposed to be used for filtering through the items..😐...I will change that now..

**Comment by MoyinoluwaA** [comment]
@ToluAkin I just fixed the first issue. Could you maybe share a screenshot for me to understand the second issue..I can't see it here. Thanks.

**Comment by ToluAkin** [comment]
> @ToluAkin I just fixed the first issue. Could you maybe share a screenshot for me to understand the second issue..I can't see it here. Thanks.

https://monosnap.com/direct/AVIx1EG3dz0YIh5oEY9LrQyrs1Okl0

**Comment by MoyinoluwaA** [comment]
@ToluAkin I saw the error in the link you sent. I corrected it now. Please review 😊

**Review by ToluAkin — APPROVED** [review]

**Pull request merged**